Description: Fix incorrect initrd minimum address calculation
 Don't add the initrd size to addr_min, since the initrd will be allocated
 after this address.  Fixes failure to load initrd on systems where the
 64MiB-128MiB range of physical memory is reserved, such as Windows Azure
 guests.
Author: Colin Watson <cjwatson@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1055686
Forwarded: https://lists.gnu.org/archive/html/grub-devel/2012-10/msg00023.html
Last-Update: 2012-10-14

Index: b/grub-core/loader/i386/linux.c
===================================================================
--- a/grub-core/loader/i386/linux.c
+++ b/grub-core/loader/i386/linux.c
@@ -1144,8 +1144,7 @@
      worse than that of Linux 2.3.xx, so avoid the last 64kb.  */
   addr_max -= 0x10000;
 
-  addr_min = (grub_addr_t) prot_mode_target + prot_init_space
-             + page_align (size);
+  addr_min = (grub_addr_t) prot_mode_target + prot_init_space;
 
   /* Put the initrd as high as possible, 4KiB aligned.  */
   addr = (addr_max - size) & ~0xFFF;
